internal/distro/fedora/images.go

```diff
@@ -129,6 +129,21 @@ func osCustomizations(
 		)
 	}
 
+	var err error
+	osc.Directories, err = blueprint.DirectoryCustomizationsToFsNodeDirectories(c.GetDirectories())
+	if err != nil {
+		// In theory this should never happen, because the blueprint directory customizations
+		// should have been validated before this point.
+		panic(fmt.Sprintf("failed to convert directory customizations to fs node directories: %v", err))
```

Comment:
I guess it's fine to panic because this would be a programming error. But why not propagate the error? All functions which invoke osCustomizations can return an error (but none do atm). Maybe the panic would need to happen higher up? idk, not a blocker any way, just wondering.

Reply:
I didn't return the error, because as you mentioned, the `osCustomizations()` does not return any. Personally I prefer to return errors instead of panicking. These customizations were the only ones that could return error while converting them. I think that we may end up returning errors also for other customizations if we start validating them.

I would be happy to rework this to return an error... 😉

Reply:
Maybe in a followup though :D